File: apps/go/routes/body/text.go
//go:build js && wasm

package main

import "github.com/primate-run/go/route"

var _ = route.With{ContentType: route.Text}.Post(func(request route.Request) any {
	s, err := request.Body.Text()
	if err != nil {
		return map[string]any{"error": err.Error()}
	}
	return s
})

File: apps/vue/routes/route-client/text.ts
import response from "primate/response";
import route from "primate/route";

export default route({
  get(request) {
    const toplevel = request.query.has("top-level") ? "TopLevel/" : "";
    return response.view(`RouteClient/${toplevel}Text.vue`);
  },
  post: route.with({ contentType: "text/plain" }, request => {
    return request.body.text();
  }),
});
